iceoryx2-bb/cxx/include/iox2/container/static_string.hpp

```diff
@@ -477,18 +539,132 @@ class StaticString {
+    ///         character sequence
+    ///         nullopt if no character is found
+    template <typename T>
+    auto find_last_of(T const& str, SizeType pos = N) const -> IsStaticStringOrCharArray<T, Optional<SizeType>> {
+        if (m_size == 0) {
+            return nullopt;
+        }
+
+        auto position = std::min(pos, m_size - 1);
+        auto str_data = detail::GetData<T>::call(str);
+        auto str_size = detail::GetSize<T>::call(str);
+        for (; position > 0; --position) {
+            auto found = memchr(str_data, m_string[position], str_size);
+            if (found != nullptr) {
+                return position;
+            }
+        }
+        auto found = memchr(str_data, m_string[0], str_size);
+        if (found != nullptr) {
+            return 0U;
+        }
+        return nullopt;
+    }
+
+    /// Inserts a StaticString, obtained by str.substr(s_index, count), into the StaticString at position index. The
+    /// insertion fails if the capacity would be exceeded or the provided indices are larger than the respective
+    /// string lengths.
+    ///
+    /// @return true if the insertion was successful, otherwise false
+    template <typename T>
+    auto insert(SizeType index, T const& str, SizeType s_index, SizeType count = T::capacity()) ->
+        typename std::enable_if_t<IsStaticString<T>::value, bool> {
+        auto sub_str = str.substr(s_index, count);
+        if (!sub_str.has_value()) {
+            return false;
+        }
+
+        auto const sub_str_size = sub_str->size();
+        auto const new_size = m_size + sub_str_size;
+        // check if the new size would exceed capacity or a size overflow occured
+        if (new_size > N || new_size < m_size) {
+            return false;
+        }
+
+        if (index > m_size) {
+            return false;
+        }
+        auto number_of_characters_to_move = m_size - index;
+        std::memmove(&m_string[index + sub_str_size], &m_string[index], number_of_characters_to_move);
+        std::memcpy(&m_string[index], sub_str->m_string, static_cast<size_t>(sub_str_size));
+
+        m_string[new_size] = '\0';
+        m_size = new_size;
+
+        return true;
+    }
+
   private:
     auto is_valid_next(char character) const noexcept -> bool {
         constexpr char const CODE_UNIT_UPPER_BOUND = 127;
         return (character > 0) && (character <= CODE_UNIT_UPPER_BOUND);
     }
-};
 
-template <typename>
-struct IsStaticString : std::false_type { };
+    auto compare(StaticString const& other) const -> int64_t {
+        auto const other_size = other.size();
+        auto const res = memcmp(&m_string[0], &other.m_string[0], std::min(m_size, other_size));
+        if (res == 0) {
+            if (m_size < other_size) {
+                return -1;
+            }
+            return ((m_size > other_size) ? 1 : 0);
+        }
+        return res;
```

Comment:
A not-so-obvious property of the UTF-8 encoding is that this actually orders strings lexicographically _by code points_. This may or may not be worth pointing out in a comment, but I think it's pretty cool.